### PR TITLE
[SPARK-38479][PYTHON] Add `Series.duplicated` to indicate duplicate Series values

### PR DESCRIPTION
**Pull request by xinrong-meng**
### What changes were proposed in this pull request?
Add `Series.duplicated` to indicate duplicate Series values.

### Why are the changes needed?
To reach parity with pandas API.

### Does this PR introduce _any_ user-facing change?
Yes, `Series.duplicated` is supported now.
```py
        >>> animals = ps.Series(['lama', 'cow', 'lama', 'beetle', 'lama'])
        >>> animals.duplicated().sort_index()
        0    False
        1    False
        2     True
        3    False
        4     True
        dtype: bool

        >>> animals.duplicated(keep='last').sort_index()
        0     True
        1    False
        2     True
        3    False
        4    False
        dtype: bool

        >>> animals.duplicated(keep=False).sort_index()
        0     True
        1    False
        2     True
        3    False
        4     True
        dtype: bool
```

### How was this patch tested?
Unit tests.

### CODE REVIEW

File: python/pyspark/pandas/tests/test_series.py [hunk 1]
```diff
@@ -387,6 +387,14 @@ def test_isin(self):
             )
             self.assert_eq(psser.isin([1, 5, 0, None]), expected)
 
+    def test_duplicated(self):
```
Comment (itholic) [inline python/pyspark/pandas/tests/test_series.py:390]: Would you mind also adding tests for numeric, Timestamp types and null cases ??

File: python/pyspark/pandas/series.py [hunk 2]
```diff
@@ -1603,6 +1603,83 @@ def to_list(self) -> List:
...
+
+        >>> animals.duplicated(keep=False).sort_index()
+        0     True
+        1    False
+        2     True
+        3    False
+        4     True
+        dtype: bool
+        """
+        return self._psdf[[self.name]].duplicated(keep=keep)
```
Comment (itholic) [inline python/pyspark/pandas/series.py:1681]: I think we should keep the name ?

e.g.

```python
name = self.name
result = self._psdf[[name]].duplicated(keep=keep)
return result.rename(name)
```

**Review by itholic — COMMENTED** [review]
Otherwise, Looks pretty good

File: python/pyspark/pandas/series.py (hunk 2 above)
Comment (xinrong-databricks) [reply python/pyspark/pandas/series.py]: Good catch! Updated.

File: python/pyspark/pandas/tests/test_series.py (hunk 1 above)
Comment (xinrong-databricks) [reply python/pyspark/pandas/tests/test_series.py:390]: Good idea, more test cases are added.

**Comment by xinrong-databricks** [comment]
Also CC @ueshin @HyukjinKwon thanks!

File: python/pyspark/pandas/missing/series.py [hunk 3]
```diff
@@ -59,7 +59,6 @@ class MissingPandasLikeSeries:
 
     # Properties we won't support.
     array = common.array(_unsupported_property)
-    duplicated = common.duplicated(_unsupported_property)
```
Comment (xinrong-databricks) [inline python/pyspark/pandas/missing/series.py:62]: There is no `duplicated` property as https://pandas.pydata.org/docs/reference/api/pandas.Series.html. It is a method actually.

File: python/pyspark/pandas/tests/test_series.py [hunk 4]
```diff
@@ -387,6 +387,28 @@ def test_isin(self):
...
+                name="times",
+            ),
+        ]:
+            psser = ps.from_pandas(pser)
+            self.assert_eq(psser.duplicated().sort_index(), pser.duplicated())
+            self.assert_eq(
+                psser.duplicated(keep="first").sort_index(), pser.duplicated(keep="first")
+            )
+            self.assert_eq(psser.duplicated(keep="last").sort_index(), pser.duplicated(keep="last"))
+            self.assert_eq(psser.duplicated(keep=False).sort_index(), pser.duplicated(keep=False))
```
Comment (ueshin) [inline python/pyspark/pandas/tests/test_series.py:410]: Shall we add a case like `(psser + 1).duplicated()`?

Comment (xinrong-databricks) [reply python/pyspark/pandas/tests/test_series.py:410]: Sure added. Then I realized it must have the same result with `psser.duplicated()` always.. It's still good to have it tested though :)

File: python/pyspark/pandas/series.py [hunk 5]
```diff
@@ -1603,6 +1603,83 @@ def to_list(self) -> List:
 
     tolist = to_list
 
+    def duplicated(self, keep: str = "first") -> "Series":
```
Comment (Yikun) [inline python/pyspark/pandas/series.py:1606]:
```suggestion
    def duplicated(self, keep: Union[bool, str] = "first") -> "Series":
```

Looks like `keep` can be str or bool, I also submit a fix on other wrong hint https://github.com/apache/spark/pull/35920

Comment (xinrong-databricks) [reply python/pyspark/pandas/series.py]: Thanks! Updated

Comment (xinrong-databricks) [reply python/pyspark/pandas/series.py]: Thanks! Updated.

File: python/pyspark/pandas/series.py [hunk 6]
```diff
@@ -1647,6 +1647,83 @@ def to_list(self) -> List:
...
+
+        Returns
+        -------
+        Series
+            Series indicating whether each value has occurred in the
+            preceding values
+
+        See Also
+        --------
+        Index.duplicated : Equivalent method on pandas.Index.
```
Comment (Yikun) [inline python/pyspark/pandas/series.py:1676]: Sorry, I missed this before, this is not supported in pyspark yet.

**Review by Yikun — COMMENTED** [review]
Sorry, I missed a doc problem before otherwise LGTM

File: python/pyspark/pandas/series.py (hunk 6 above)
Comment (xinrong-databricks) [reply python/pyspark/pandas/series.py:1676]: Good catch! Modified.

**Comment by xinrong-databricks** [comment]
Rebased for latest type hints changes; no new code changes related to this PR after review.

**Comment by HyukjinKwon** [comment]
Merged to master

File: python/pyspark/pandas/series.py [hunk 7]
```diff
@@ -1647,6 +1647,83 @@ def to_list(self) -> List:
 
     tolist = to_list
 
+    def duplicated(self, keep: Union[bool, str] = "first") -> "Series":
```
Comment (HyukjinKwon) [inline python/pyspark/pandas/series.py:1650]: Let;s add `versionadded: 3.4.0`

**Review by HyukjinKwon — APPROVED** [review]
LGTM2 otherwise

File: python/pyspark/pandas/series.py (hunk 7 above)
Comment (xinrong-databricks) [reply python/pyspark/pandas/series.py:1650]: Makes sense, added!

**Comment by HyukjinKwon** [comment]
Merged to master.
